integrations/nielsen-dtvr/lib/index.js

```diff
@@ -0,0 +1,347 @@
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Buffer Started
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackBufferStarted = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Buffer Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackBufferCompleted = function(track) {
+  var metadata = this.isAd ? this.mapAd(track) : this.mapVideo(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Paused
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackPaused = function(track) {
+  this.sendID3(track);
+  this.end(track);
+};
+
+/**
+ * Video Playback Resumed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackResumed = function(track) {
+  var metadata = this.isAd ? this.mapAd(track) : this.mapVideo(track);
+  this.client.ggPM('loadMetadata', metadata);
+  this.sendID3(track);
+};
+
+/**
+ * Video Playback Completed
+ *
+ * @api public
+ */
+
+NielsenDTVR.prototype.videoPlaybackCompleted = function(track) {
+  this.end(track);
+};
+
+/**
+ * Send ID3 tags to Nielsen
+ *
+ * @api private
+ */
+
+NielsenDTVR.prototype.sendID3 = function(event) {
+  var id3Events = this.options.sendId3Events;
+  var i;
+  var id3Prop;
+  var id3Tags;
+
+  if (id3Events.length <= 0) return;
+  if (id3Events.length > 0) {
+    for (i = 0; i < id3Events.length; i += 1) {
+      if (id3Events[i] === event.event()) {
+        break;
+      }
+      if (i === id3Events.length - 1) {
+        return;
+      }
+    }
+  }
+
+  id3Prop = this.options.id3Property || 'ID3';
+  id3Tags = event.proxy('properties.' + id3Prop);
+  if (id3Tags) {
+    // we'll only send ID3 tags to Nielsen if we detect the customer has either
+    // never sent ID3 tags, or if they're different from the previous ID3 tags
+    // sent during the session
+    if (!this.ID3) {
+      this.ID3 = id3Tags;
+      this.client.ggPM('sendID3', this.ID3);
+    } else if (id3Tags !== this.ID3) {
+      this.client.ggPM('sendID3', this.ID3);
+    }
+  }
+};
+
+/**
+ * End playback
+ *
+ * @api private
+ */
+
+NielsenDTVR.prototype.end = function(event) {
+  var livestream = event.proxy('properties.livestream');
+  var position = event.proxy('properties.position');
+  var time;
+  if (livestream) {
+    time = Date.now(event.timestamp());
+  } else if (position) {
+    time = position;
+  }
+
+  this.client.ggPM('end', time);
+};
+
+/**
+ * Map video properties
+ *
+ * @api private
+ */
+
+NielsenDTVR.prototype.mapVideo = function(event) {
+  var loadType;
+  var loadTypeVal =
+    event.proxy('properties.loadType') || event.proxy('properties.load_type');
+
+  if (loadTypeVal === 'linear') {
+    loadType = '1';
+  } else if (loadTypeVal === 'dynamic') {
+    loadType = '2';
+  }
+
+  return reject({
+    type: 'content',
+    channelName: event.proxy('properties.channel'),
```

Comment:
Is channel name required?

Reply:
Yes, added a helper method so that Segment does not send a `loadMetadata` event if all required properties are not present.